jax/interpreters/numpy_eval.py

```diff
@@ -0,0 +1,615 @@
+                            np.remainder(lhs, rhs) != 0)
+    return np.where(select, quotient + 1, quotient)
+  else:
+    return np.divide(lhs, rhs)
+np_impl[lax.div_p] = div
+
+def rem(lhs, rhs):
+  return np.sign(lhs) * np.remainder(np.abs(lhs), np.abs(rhs))
+np_impl[lax.rem_p] = rem
+
+max = np.maximum
+np_impl[lax.max_p] = max
+
+min = np.minimum
+np_impl[lax.min_p] = min
+
+
+shift_left = np.left_shift
+np_impl[lax.shift_left_p] = shift_left
+
+shift_right_arithmetic = np.right_shift
+np_impl[lax.shift_right_arithmetic_p] = shift_right_arithmetic
+
+_logical_types = {
+  np.dtype('uint8'): np.uint8,
+  np.dtype('uint16'): np.uint16,
+  np.dtype('uint32'): np.uint32,
+  np.dtype('uint64'): np.uint64,
+  np.dtype('int8'): np.uint8,
+  np.dtype('int16'): np.uint16,
+  np.dtype('int32'): np.uint32,
+  np.dtype('int64'): np.uint64,
+  np.dtype('float16'): np.uint16,
+  np.dtype('float32'): np.uint32,
+  np.dtype('float64'): np.uint64,
+}
+
+def shift_right_logical(x1, x2):
+  x1 = np.asarray(x1)
+  logical_type = _logical_types[x1.dtype]
+  shifted = np.right_shift(x1.view(logical_type), x2)
+  return shifted.astype(logical_type).view(x1.dtype)
+np_impl[lax.shift_right_logical_p] = shift_right_logical
+
+def population_count(x):
+  assert np.issubdtype(x.dtype, np.integer)
+  dtype = x.dtype
+  iinfo = np.iinfo(x.dtype)
+  if np.iinfo(x.dtype).bits < 32:
+    assert iinfo.kind in ('i', 'u')
+    x = x.astype(np.uint32 if iinfo.kind == 'u' else np.int32)
+  if iinfo.kind == 'i':
+    x = x.view(f"uint{np.iinfo(x.dtype).bits}")
+  assert x.dtype in (np.uint32, np.uint64)
+  m = [
+    0x5555555555555555,  # binary: 0101...
+    0x3333333333333333,  # binary: 00110011..
+    0x0f0f0f0f0f0f0f0f,  # binary:  4 zeros,  4 ones ...
+    0x00ff00ff00ff00ff,  # binary:  8 zeros,  8 ones ...
+    0x0000ffff0000ffff,  # binary: 16 zeros, 16 ones ...
+    0x00000000ffffffff,  # binary: 32 zeros, 32 ones
+  ]
+
+  if x.dtype == np.uint32:
+    m = list(map(np.uint32, m[:-1]))
+  else:
+    m = list(map(np.uint64, m))
+
+  x = (x & m[0]) + ((x >>  1) & m[0])  # put count of each  2 bits into those  2 bits
+  x = (x & m[1]) + ((x >>  2) & m[1])  # put count of each  4 bits into those  4 bits
+  x = (x & m[2]) + ((x >>  4) & m[2])  # put count of each  8 bits into those  8 bits
+  x = (x & m[3]) + ((x >>  8) & m[3])  # put count of each 16 bits into those 16 bits
+  x = (x & m[4]) + ((x >> 16) & m[4])  # put count of each 32 bits into those 32 bits
+  if x.dtype == np.uint64:
+    x = (x & m[5]) + ((x >> 32) & m[5])  # put count of each 64 bits into those 64 bits
+  return x.astype(dtype)
+np_impl[lax.population_count_p] = population_count
+
+eq = np.equal
+np_impl[lax.eq_p] = eq
+
+ne = np.not_equal
+np_impl[lax.ne_p] = ne
+
+ge = np.greater_equal
+np_impl[lax.ge_p] = ge
+
+gt = np.greater
+np_impl[lax.gt_p] = gt
+
+le = np.less_equal
+np_impl[lax.le_p] = le
+
+lt = np.less
+np_impl[lax.lt_p] = lt
+
+def convert_element_type(operand, dtype, old_dtype=None):
```

Comment:
I think you need to rename `dtype` -> `new_dtype` to match lax https://github.com/google/jax/blob/ea6d1dafc2f698a831cfdbb7faf1f4dab47ed102/jax/lax/lax.py#L368 (this was causing a failure in FastAR).

Reply:
Actually is it possible for the tests to assert that the signatures of these functions are correct?

Reply:
@j-towns Sorry, now fixed. The tests are catching this. 

However, this was shadowed by a (recurrent/sporadic) build failure due to TF package imports on a recent run. @mattjj These are annoying, I had to restart every 2nd to 3rd run because of this. Can we fix this?